### ISSUE TITLE
Recursive WKT imports not properly linked

### ISSUE THREAD

**Issue by pedgeio** [jhump/protoreflect#219]
Will provide better reproduction steps as I get them, but I thought you might immediately know the issue.

https://github.com/protocolbuffers/protobuf/blob/b008801b5d10fd06b3a80c6f88a01f9db73c016b/src/google/protobuf/type.proto#L57
https://github.com/protocolbuffers/protobuf/blob/b008801b5d10fd06b3a80c6f88a01f9db73c016b/src/google/protobuf/source_context.proto#L44

`google.protobuf.Type` uses `google.protobuf.SourceContext`, but if you use `google.protobuf.Type` in your own code, and call `protoparse.Parser#ParseFiles(...)`, you will get:

```
google/protobuf/type.proto: field google.protobuf.Type.source_context: unknown type .google.protobuf.SourceContext
```

Originally I thought the issue was in `parseProtoFiles` where it appeared to only do one-depth for dependencies for `standardImports`, but if `recursive == true` as it does when called from `ParseFiles`, the dependencies (namely `google/protobuf/source_context.proto`) should be added to `parsed`. I'll work to come up with a proper test for this.

**Comment by pedgeio** [comment]
Actually it seems as if the file is not included. From the root of googleapis:

```
$ goprotoc -I . google/api/*.proto
google/protobuf/type.proto: field google.protobuf.Type.source_context: unknown type .google.protobuf.SourceContext
```

If I do some print statements from `protoparse`:

```diff
diff --git a/desc/protoparse/parser.go b/desc/protoparse/parser.go
index ce9a3e4..b26a2b4 100644
--- a/desc/protoparse/parser.go
+++ b/desc/protoparse/parser.go
@@ -225,8 +225,18 @@ func (p Parser) ParseFiles(filenames ...string) ([]*desc.FileDescriptor, error)
        if p.InferImportPaths {
                protos = fixupFilenames(protos)
        }
+       var names []string
+       for name := range protos {
+               names = append(names, name)
+       }
+       sort.Strings(names)
+       for _, name := range names {
+               fmt.Fprintln(os.Stderr, name)
+       }
+       fmt.Fprintln(os.Stderr)
        linkedProtos, err := newLinker(protos).linkFiles()
        if err != nil {
+               fmt.Fprintln(os.Stderr, "HERE")
                return nil, err
        }
        if p.IncludeSourceCodeInfo {
```

It shows that `google/protobuf/source_context.proto` is not included:

```
google/api/annotations.proto
google/api/auth.proto
google/api/backend.proto
google/api/billing.proto
google/api/client.proto
google/api/config_change.proto
google/api/consumer.proto
google/api/context.proto
google/api/control.proto
google/api/distribution.proto
google/api/documentation.proto
google/api/endpoint.proto
google/api/field_behavior.proto
google/api/http.proto
google/api/httpbody.proto
google/api/label.proto
google/api/launch_stage.proto
google/api/log.proto
google/api/logging.proto
google/api/metric.proto
google/api/monitored_resource.proto
google/api/monitoring.proto
google/api/quota.proto
google/api/resource.proto
google/api/service.proto
google/api/source_info.proto
google/api/system_parameter.proto
google/api/usage.proto
google/protobuf/any.proto
google/protobuf/api.proto
google/protobuf/descriptor.proto
google/protobuf/duration.proto
google/protobuf/struct.proto
google/protobuf/timestamp.proto
google/protobuf/type.proto
google/protobuf/wrappers.proto

HERE
google/protobuf/api.proto: field google.protobuf.Api.source_context: unknown type .google.protobuf.SourceContext
```

**Comment by pedgeio** [comment]
Fix in #220
